testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/RequiredActionEmailVerificationTest.java

```diff
@@ -868,6 +877,45 @@ public void verifyEmailWhileLoggedIn() throws IOException, MessagingException {
         Assert.assertThat(user.getRequiredActions(), Matchers.empty());
     }
 
+    @Test
+    public void verifyEmailViaAuthSessionWhileLoggedIn() throws IOException, MessagingException {
+        UserAttributeUpdater userAttributeUpdater = new UserAttributeUpdater(testRealm().users().get(testUserId));
+        userAttributeUpdater.setEmailVerified(false).update();
+
+        final String testRealmName = testRealm().toRepresentation().getRealm();
+        accountPage.setAuthRealm(testRealmName);
+        oauth.realm(testRealmName).clientId("account").redirectUri(getAuthServerRoot() + "realms/" + testRealmName + "/account");
+        loginPage.open();
+
+        String authSessionId = AuthenticationSessionFailoverClusterTest.getAuthSessionCookieValue(driver);
```

Comment:
Did you test on Wildfly? Not 100% sure, but AFAIR this probably won't work on Wildfly (authSessionId and realmId may not be visible in the inner block, which is executed on completely different VM than test).

I think for long term perspective, it will be good to have some better support for transfer the state between test and VM. I think that ATM it is not so well supported...

Reply:
You're right, I've introduced https://issues.jboss.org/browse/KEYCLOAK-8863 to address it and the tests would be then updated after that one is implemented.